iroh/src/config.rs

```diff
@@ -22,6 +23,55 @@ pub const CONFIG_FILE_NAME: &str = "iroh.config.toml";
 /// For example, `IROH_PATH=/path/to/config` would set the value of the `Config.path` field
 pub const ENV_PREFIX: &str = "IROH";
 
+/// Paths to files or directory within the [`iroh_data_root`] used by Iroh.
+pub enum IrohPaths {
+    /// Path to the node's private key for the [`crate::net::PeerId`]
+    Keypair,
+    /// Path to the node's flat-file bao blob store ([`crate::baomap::flat`])
+    BaoFlatStore,
+}
+impl From<&IrohPaths> for &'static str {
+    fn from(value: &IrohPaths) -> Self {
+        match value {
+            IrohPaths::Keypair => "keypair",
+            IrohPaths::BaoFlatStore => "blobs.v0",
+        }
+    }
+}
+impl FromStr for IrohPaths {
+    type Err = anyhow::Error;
+    fn from_str(s: &str) -> Result<Self> {
+        Ok(match s {
+            "keypair" => Self::Keypair,
+            "blobs.v0" => Self::BaoFlatStore,
+            _ => bail!("unknown file or directory"),
```

Comment:
would be good to have a basic test for this

Reply:
added a encode-parse roundtrip test